### PR TITLE
Sandbox login via onboarding wizard & refactoring (3897)

### PR DESCRIPTION
**Pull request by stracker-phil**
### Description

_Massive improvements to the new settings UI_

**Overview**

This PR primarily focuses on backend enhancements, including:
- Complete overhaul of the Redux store
- Standardization of REST endpoints
- Implementation of initial logic for sandbox login popup

### CODE REVIEW

**Review by danieldudzic — APPROVED** [review]
LGTM!

File: modules/ppcp-settings/src/Data/CommonSettings.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	protected function get_defaults() : array {
+		return array(
+			'use_sandbox'           => false,
+			'use_manual_connection' => false,
+			'client_id'             => '',
+			'client_secret'         => '',
+		);
+	}
+
+	// -----
```
Comment (Narek13) [inline modules/ppcp-settings/src/Data/CommonSettings.php:46]: leftover?

File: modules/ppcp-settings/src/Endpoint/CommonRestEndpoint.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ * This API acts as the intermediary between the "external world" and our
+ * internal data model.
+ */
+class CommonRestEndpoint extends RestEndpoint {
+	/**
+	 * The base path for this REST controller.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'common';
```
Comment (Narek13) [inline modules/ppcp-settings/src/Endpoint/CommonRestEndpoint.php]:
```suggestion
	protected string $rest_base = 'common';
```

File: modules/ppcp-settings/src/Endpoint/LoginLinkRestEndpoint.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+/**
+ * REST controller that generates merchant login URLs.
+ */
+class LoginLinkRestEndpoint extends RestEndpoint {
+	/**
+	 * The base path for this REST controller.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'login_link';
```
Comment (Narek13) [inline modules/ppcp-settings/src/Endpoint/LoginLinkRestEndpoint.php]:
```suggestion
	protected string $rest_base = 'login_link';
```
Same here, is there a reason to not use the type-hints for these specific properties?

**Review by Narek13 — APPROVED** [review]
@stracker-phil Thanks for this really massive PR. I haven't tested locally but code wise looks great to me. Just couple of suggestions which you can ignore( probably except the "leftover" one )

File: modules/ppcp-settings/src/Data/CommonSettings.php (hunk 1 above)
Comment (stracker-phil) [reply modules/ppcp-settings/src/Data/CommonSettings.php:46]: We can remove it anytime. My idea was to have a "separator" between regular class methods (top part), and getters/setters (bottom)

Comment (Narek13) [reply modules/ppcp-settings/src/Data/CommonSettings.php:46]: sure, thanks for the explanation.

**Pull request merged**